flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -81,10 +80,8 @@
 
 	private final FlinkKubeClient kubeClient;
 
-	private final ContaineredTaskManagerParameters taskManagerParameters;
-
-	/** The number of pods requested, but not yet granted. */
-	private int numPendingPodRequests = 0;
+	/** Map from pod name to worker resource. */
+	private final Map<String, WorkerResourceSpec> podWorkerResources;
```

Comment:
1. True. I'll try to clean up when the pod with corresponding name is removed from `workerNodes`. I believe that means we no longer need its resource information.

2. I think it's by design. The original behavior is that RM asks SM for pending slots, and will only request new pod only if there are more pending slots than the slots of the pending TM/pods. In case of JM failover, there should be no pending slots from previous attempt. Therefore, only slots requested in the new attempt is considered. I think the current behavior aligns with the original.